analytics/record.go

```diff
@@ -144,28 +154,75 @@ func (r Record) MarshalJSON() ([]byte, error) {
 	}
 	for _, attr := range attrs {
 		key := attr.Name
+		val := attr.Value
+
+		if err := validateAttribute(attr); err != nil {
+			log.Debugf(err.Error())
+			continue
+		}
+
+		// ensure prefix
 		if !strings.HasPrefix(key, attributePrefix) {
-			key = attributePrefix + attr.Name
+			key = attributePrefix + key
 		}
-		val := truncStringToBytes(attr.Value, maxAttributeValueBytes)
-		m[key] = val
-		if len(val) != len(attr.Value) {
+
+		// format time as int (ms since epoch)
+		if t, ok := val.(time.Time); ok {
+			val = timeToApigeeInt(t)
+		}
+
+		// truncate, if necessary
+		val, truncated := truncStringToBytes(val, maxAttributeValueBytes)
+		if truncated {
 			log.Debugf("truncated attribute %s to max length of %d", key, maxAttributeValueBytes)
 		}
+		m[key] = val
+	}
+	b, e := json.Marshal(m)
+	if e != nil {
+		log.Debugf("ax json err: %s", e)
+	} else {
+		log.Debugf("ax record: %s", string(b))
+	}
+	return b, e
+}
+
+// format time as ms since epoch
+func timeToApigeeInt(t time.Time) int64 {
+	return t.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
```

Comment:
super nit: I know this is not expected to go wrong but it could be covered in a simple test if you'd like :)